meilies-server/src/main.rs

```diff
@@ -167,11 +190,20 @@ fn handle_request(
                 }));
             }
         },
-        Request::Publish { stream, event } => {
+        Request::Publish { stream, event_name, event_data } => {
             let tree = db.open_tree(stream.into_bytes())?;
             let event_id = EventId::from(db.generate_id()?);
 
-            if let Err(e) = tree.set(event_id, event.0) {
+            let raw_length = event_name.as_str().len().to_be_bytes().to_vec();
+            let raw_name = event_name.into_bytes();
+            let raw_data = event_data.0;
```

Comment:
you do not need the `to_vec` on the first `raw_length`.
And `as_bytes` can be used for the `raw_name`.